File: day1/Day1.cs
using System;
using System.IO;
using System.Linq;
using System.Collections.Generic;

public class Globals {
    public static string input_file {get;} = "day1.input";
}

public class Program {
        public static int Main(string[] args) {
            List<string> lines = File.ReadLines(Globals.input_file).ToList();
            int[] sonar = new int[lines.Count];
            

            for(int i = 0; i < sonar.Length; i++) {
                sonar[i] = int.Parse(lines[i]);
                Console.WriteLine("{0}: {1}",i,sonar[i]);
            }
    
            int gold = 
                (from int i in Enumerable.Range(3,sonar.Length-3)
                where sonar[i] > sonar[i-3]
                select 1).Sum();
            Console.WriteLine("Gold {0}",gold);
            
            int silver =
                (from int i in Enumerable.Range(1,sonar.Length-1)
                where sonar[i] > sonar[i-1]
                select 1).Sum();
            Console.WriteLine("Silver {0}",silver);
            
            return 0;
        }
}

File: day1/day1.py
from advent import Session

aoc = Session(2021,1)
silver = None
gold = None


lines = []
with aoc.fp() as f:
    lines = [int(line.strip()) for line in f.readlines()]

count = 0
last_line = lines[0]
for line in lines[1:]:
    if line > last_line:
        count += 1
    last_line = line

silver = count

window = lines[0] + lines[1] + lines[2]
count = 0
i=3
while i < len(lines):
    new_window = window - lines[i-3] + lines[i]
    if new_window > window:
        count += 1
    window = new_window
    i += 1
    
gold = count

aoc.solution(1,silver)
aoc.solution(2,gold)
